geonode/base/models.py

```diff
@@ -676,8 +676,6 @@ class ResourceBase(PolymorphicModel, PermissionLevelMixin, ItemBase):
         help_text=purpose_help_text)
     owner = models.ForeignKey(
         settings.AUTH_USER_MODEL,
-        blank=True,
-        null=True,
         related_name='owned_resource',
         verbose_name=_("Owner"),
         on_delete=models.CASCADE)
```

Comment:
Please change `models.CASCADE` to `PROTECT` or `SET_DEFAULT`.

We don't want to risk deleting all the resources in the case we remove a user.